import sys
from PyQt5 import QtWidgets, uic

questions = [
    # 1번 ~ 6번 문제
    {
        "text": "자주 어지럽나요?",
        "options": [
            "1 (1주일에 5번 이상)",
            "2 (1주일에 3~4번)",
            "3 (1주일에 1~2번)",
            "4 (어지럽지 않다)"
        ],
        "deduct": {
            "1 (1주일에 5번 이상)": 10,
            "2 (1주일에 3~4번)": 8,
            "3 (1주일에 1~2번)": 4,
            "4 (어지럽지 않다)": 0
        },
        "advice": {
            "1 (1주일에 5번 이상)": "어지럼증이 자주 있다면, 철분이 부족할 수 있어요! 철분이 풍부한 음식(시금치, 소고기, 달걀노른자 등)을 자주 먹고, 필요하다면 어린이용 철분 영양제를 부모님과 상의해보세요. 물도 충분히 마시고, 아침을 꼭 챙겨 먹는 습관을 들이면 좋아요!",
            "2 (1주일에 3~4번)": "어지럼증이 종종 있다면, 식사와 수분 섭취를 잘 챙기고, 규칙적으로 운동해보세요! 철분, 비타민C가 풍부한 과일(딸기, 오렌지 등)도 함께 먹으면 도움이 돼요.",
            "3 (1주일에 1~2번)": "가끔 어지럽다면, 너무 걱정하지 않아도 돼요! 하지만 식사를 거르지 않고, 다양한 음식을 골고루 먹는 습관을 들이면 더 건강해질 수 있어요",
            "4 (어지럽지 않다)": "정말 훌륭해요! 건강관리가 아주 잘 되고 있어요. 지금처럼 규칙적인 식사와 운동을 계속하면 멋진 건강을 유지할 수 있어요!"
        }
    },
    {
        "text": "평소에 눈이 흔들리나요?",
        "options": [
            "1 (예)",
            "2 (아니오)"
        ],
        "deduct": {
            "1 (예)": 10,
            "2 (아니오)": 0
        },
        "advice": {
            "1 (예)": "눈이 흔들린다면, 눈 건강을 위해 비타민A와 오메가-3가 풍부한 음식을 먹어보세요! 당근, 연어, 고등어, 시금치 등이 좋아요. 눈이 피로하지 않도록 스마트폰이나 TV를 오래 보지 않도록 주의해요. 필요하다면 부모님과 함께 영양제도 상담해보세요.",
            "2 (아니오)": "눈이 흔들리지 않는다면 아주 좋아요! 지금처럼 눈 건강을 잘 지키고, 밝은 곳에서 책을 읽고, 눈을 자주 쉬게 해주세요. 당근, 달걀, 우유 등도 꾸준히 먹으면 더 좋아요!"
        }
    },
    {
        "text": "감기에 자주 걸리십니까?",
        "options": [
            "1 (1달에 2번 이하)",
            "2 (1달에 3번 이상 5번 이하)",
            "3 (1달에 6번 이상)"
        ],
        "deduct": {
            "1 (1달에 2번 이하)": 0,
            "2 (1달에 3번 이상 5번 이하)": 5,
            "3 (1달에 6번 이상)": 10
        },
        "advice": {
            "1 (1달에 2번 이하)": "면역력이 아주 좋아요! 지금처럼 과일, 채소, 단백질을 골고루 먹고, 손을 자주 씻는 습관을 계속 유지하세요.",
            "2 (1달에 3번 이상 5번 이하)": "감기에 자주 걸린다면, 비타민C와 아연이 풍부한 음식을 더 챙겨 먹어보세요! 귤, 딸기, 키위, 견과류, 해산물 등이 도움이 돼요. 필요하다면 어린이용 종합비타민도 부모님과 상의해볼 수 있어요.",
            "3 (1달에 6번 이상)": "감기에 너무 자주 걸린다면, 면역력을 키우는 것이 중요해요! 비타민C, 아연, 프로바이오틱스가 들어있는 음식(요거트, 김치, 과일 등)을 자주 먹고, 충분한 수면과 손 씻기를 꼭 실천하세요. 필요하다면 영양제도 고려해보세요."
        }
    },
    {
        "text": "피부가 건조하거나 손톱이 자주 부러지나요?",
        "options": [
            "1 (예)",
            "2 (아니오)"
        ],
        "deduct": {
            "1 (예)": 10,
            "2 (아니오)": 0
        },
        "advice": {
            "1 (예)": "피부와 손톱 건강을 위해 비타민A, 비타민E, 오메가-3가 풍부한 음식을 먹어보세요! 견과류, 연어, 아보카도, 당근, 달걀 등이 좋아요. 물도 충분히 마시고, 손톱을 깨물지 않도록 주의해요. 필요하다면 어린이용 종합비타민도 도움이 될 수 있어요.",
            "2 (아니오)": "피부와 손톱이 건강하다니 정말 멋져요! 지금처럼 다양한 음식을 골고루 먹고, 손을 깨끗이 씻는 습관을 계속 유지하세요."
        }
    },
    {
        "text": "변비에 자주 걸리시나요?",
        "options": [
            "1 (2주에 2번 이하)",
            "2 (2주에 3~4번)",
            "3 (2주에 5번)",
            "4 (2주에 6번 이상)"
        ],
        "deduct": {
            "1 (2주에 2번 이하)": 0,
            "2 (2주에 3~4번)": 4,
            "3 (2주에 5번)": 8,
            "4 (2주에 6번 이상)": 10
        },
        "advice": {
            "1 (2주에 2번 이하)": "배변 습관이 아주 좋아요! 지금처럼 물을 자주 마시고, 채소와 과일을 꾸준히 먹으면 건강한 장을 유지할 수 있어요.",
            "2 (2주에 3~4번)": "가끔 변비가 있다면, 식이섬유가 풍부한 음식(고구마, 바나나, 사과, 현미 등)을 더 챙겨 먹어보세요! 물을 충분히 마시고, 규칙적으로 운동하면 더 좋아요.",
            "3 (2주에 5번)": "변비가 자주 있다면, 식이섬유와 유산균이 풍부한 음식을 꼭 챙겨 먹고, 화장실에 가고 싶을 때 참지 마세요! 요거트, 김치, 바나나, 오트밀 등이 도움이 돼요. 필요하다면 어린이용 유산균 영양제도 고려해보세요.",
            "4 (2주에 6번 이상)": "변비가 너무 자주 있다면, 식습관을 꼭 바꿔야 해요! 채소, 과일, 통곡물, 유산균이 풍부한 음식을 많이 먹고, 물을 자주 마시세요. 운동도 꾸준히 하면 장 건강에 큰 도움이 돼요. 필요하다면 부모님과 함께 유산균을 비롯한 영양제도 상담해보세요."
        }
    },

    # 7번 ~ 13번 문제 
    {
        "text": "하루에 평균 몇 시간을 자나요?",
        "options": [
            "1 (5시간 미만)",
            "2 (5시간~7시간 미만)",
            "3 (7시간~8시간 미만)",
            "4 (8시간 이상)"
        ],
        "deduct": {
            "1 (5시간 미만)": 10,
            "2 (5시간~7시간 미만)": 8,
            "3 (7시간~8시간 미만)": 6,
            "4 (8시간 이상)": 0
        },
        "advice": {
            "1 (5시간 미만)": "수면이 너무 부족해요! 성장기에는 하루에 8~9시간 정도 자는 것이 가장 좋아요. 잠이 부족하면 키 성장, 집중력, 면역력에 모두 나쁜 영향을 줄 수 있어요. 밤에는 스마트폰이나 TV를 일찍 끄고, 규칙적으로 같은 시간에 자는 습관을 들여보세요. 만약 잠이 잘 오지 않는다면, 자기 전 따뜻한 우유를 마시거나 가벼운 스트레칭을 해보는 것도 좋아요. 충분한 수면은 건강한 몸과 똑똑한 두뇌를 만들어줘요!",
            "2 (5시간~7시간 미만)": "잠이 조금 부족한 편이에요! 초등학생은 하루에 8~9시간 이상 자야 키도 쑥쑥 크고, 학교 생활도 힘차게 할 수 있어요.늦게까지 공부하거나 게임을 하느라 잠을 줄이지 않도록 주의하고, 자기 전에는 밝은 화면을 멀리하세요. 규칙적인 수면 습관을 들이면 더 건강해질 수 있어요!",
            "3 (7시간~8시간 미만)": "조금만 더 자면 더 건강해질 수 있어요! 8시간 이상 자는 것이 가장 좋답니다.자기 전에는 휴대폰 대신 책을 읽거나 조용한 음악을 듣는 것도 좋아요. 잠을 충분히 자면 아침에 더 상쾌하게 일어날 수 있어요!",
            "4 (8시간 이상)": "아주 잘하고 있어요! 성장기에는 8~9시간 정도 자는 것이 가장 좋아요. 지금처럼 규칙적으로 충분히 자면 키도 잘 크고, 집중력도 좋아져서 학교 생활이 훨씬 즐거워질 거예요. 앞으로도 건강한 수면 습관을 계속 지켜주세요!"
        }
    },
    {
        "text": "아침을 매일 먹나요?",
        "options": [
            "1 (예)",
            "2 (아니오)"
        ],
        "deduct": {
            "1 (예)": 0,
            "2 (아니오)": 10
        },
        "advice": {
            "1 (예)": "아주 잘하고 있어요! 아침을 매일 먹으면 뇌가 깨어나고, 학교에서 집중력도 높아져요. 성장기에는 아침 식사가 정말 중요해요! 밥, 달걀, 과일, 우유 등 다양한 음식을 골고루 먹으면 더 건강해질 수 있어요. 앞으로도 아침을 꼭 챙겨 먹는 좋은 습관을 계속 이어가세요!",
            "2 (아니오)": "아침을 거르면 힘이 없고, 공부에도 집중하기 어려워질 수 있어요! 아침을 먹지 않으면 영양이 부족해져서 성장에도 방해가 될 수 있답니다. 바쁠 때는 바나나, 우유, 삶은 달걀처럼 간단한 음식이라도 꼭 먹어보세요. 아침을 챙겨 먹는 습관이 건강과 키 성장, 활기찬 하루를 만들어줘요! 내일부터라도 아침을 꼭 챙겨 먹어보는 건 어떨까요? 화이팅!"
        }
    },
    {
        "text": "과일이나 야채를 얼마나 섭취하나요?",
        "options": [
            "1 (전혀 안 먹는다)",
            "2 (하루 1회)",
            "3 (하루 2~3회)",
            "4 (하루 4회 이상)"
        ],
        "deduct": {
            "1 (전혀 안 먹는다)": 10,
            "2 (하루 1회)": 8,
            "3 (하루 2~3회)": 4,
            "4 (하루 4회 이상)": 0
        },
        "advice": {
            "1 (전혀 안 먹는다)": "과일과 야채를 전혀 먹지 않으면 비타민과 식이섬유가 부족해질 수 있어요! 비타민과 식이섬유는 키 성장, 피부 건강, 면역력에 아주 중요해요. 매일 조금씩이라도 과일(사과, 바나나, 딸기 등)과 야채(당근, 오이, 브로콜리 등)를 먹는 습관을 들여보세요. 처음에는 좋아하는 것부터 시작해도 괜찮아요! 필요하다면 어린이용 종합비타민도 도움이 될 수 있으니 부모님과 상의해보세요.",
            "2 (하루 1회)": "과일이나 야채를 조금만 더 자주 먹으면 더 건강해질 수 있어요! 하루에 2~3번 이상, 다양한 색깔의 과일과 야채를 골고루 먹어보세요. 샐러드, 과일 주스, 야채 스틱 등 맛있게 먹는 방법도 많아요. 식사 때마다 조금씩 추가해보는 것도 좋은 방법이에요!",
            "3 (하루 2~3회)": "좋은 습관이에요! 하루 2~3번 과일과 야채를 챙겨 먹으면 비타민, 미네랄, 식이섬유를 충분히 섭취할 수 있어요. 앞으로도 다양한 과일과 야채를 골고루 먹는 습관을 계속 이어가면 더 건강해질 거예요!",
            "4 (하루 4회 이상)": "정말 멋져요! 하루 4번 이상 과일과 야채를 먹으면 건강에 필요한 영양소를 충분히 섭취할 수 있어요. 지금처럼 다양한 색깔의 과일과 야채를 골고루 먹는 습관을 계속 이어가세요. 건강도 튼튼, 기분도 상쾌해질 거예요!"
        }
    },
    {
        "text": "패스트푸드나 탄산음료를 얼마나 섭취하나요?",
        "options": [
            "1 (거의 매일)",
            "2 (주 3~4회)",
            "3 (주 1~2회)",
            "4 (거의 안 먹는다)"
        ],
        "deduct": {
            "1 (거의 매일)": 10,
            "2 (주 3~4회)": 8,
            "3 (주 1~2회)": 4,
            "4 (거의 안 먹는다)": 0
        },
        "advice": {
            "1 (거의 매일)": "패스트푸드와 탄산음료를 거의 매일 먹으면 건강에 큰 해가 될 수 있어요! 이런 음식은 지방, 당분, 염분이 많고 비타민과 미네랄이 부족해서 비만, 성적 저하, 집중력 저하, 심지어 ADHD 위험까지 높아질 수 있어요. 건강한 성장과 똑똑한 두뇌를 위해 패스트푸드와 탄산음료는 특별한 날에만 먹고, 평소에는 집밥, 과일, 채소, 물을 더 많이 먹는 습관을 들여보세요!",
            "2 (주 3~4회)": "패스트푸드와 탄산음료를 자주 먹는 편이군요! 이런 식습관은 영양 불균형과 비만, 성적 저하, 나쁜 식습관으로 이어질 수 있어요.주 1회 이하로 줄이고, 간식으로는 과일, 견과류, 요거트, 우유 등을 선택해보세요. 물을 자주 마시고, 집에서 만든 건강한 음식을 더 많이 먹는 습관을 길러보세요!",
            "3 (주 1~2회)": "가끔 먹는 건 괜찮지만, 더 줄이면 더 건강해질 수 있어요! 패스트푸드와 탄산음료 대신 과일, 채소, 집밥을 더 자주 먹으면 몸도 마음도 튼튼해져요. 특별한 날에만 즐기고, 평소에는 건강한 음식을 골고루 먹는 습관을 계속 이어가면 좋아요!",
            "4 (거의 안 먹는다)": "정말 멋져요! 건강을 아주 잘 지키고 있네요! 패스트푸드와 탄산음료를 거의 먹지 않으면 비만, 성적 저하, 만성질환 위험이 줄어들고, 건강하게 성장할 수 있어요. 지금처럼 집밥, 과일, 채소, 물을 자주 먹는 습관을 계속 지켜주세요!"
        }
    },
    {
        "text": "등 푸른 생선을 얼마나 자주 먹나요?",
        "options": [
            "1 (일주일에 한 번도 안 먹는다)",
            "2 (일주일에 1번 먹는다)",
            "3 (일주일에 2~3번 먹는다)",
            "4 (일주일에 4번 이상 먹는다)"
        ],
        "deduct": {
            "1 (일주일에 한 번도 안 먹는다)": 10,
            "2 (일주일에 1번 먹는다)": 6,
            "3 (일주일에 2~3번 먹는다)": 3,
            "4 (일주일에 4번 이상 먹는다)": 0
        },
        "advice": {
            "1 (일주일에 한 번도 안 먹는다)": "생선을 거의 먹지 않는다면 오메가-3와 비타민D가 부족할 수 있어요! 등푸른 생선(고등어, 연어, 참치 등)은 두뇌 발달과 눈 건강에 아주 좋아요. 일주일에 2~3번은 생선을 먹는 습관을 들여보세요. 생선을 잘 못 먹겠다면, 어린이용 오메가-3 영양제도 부모님과 상의해볼 수 있어요.",
            "2 (일주일에 1번 먹는다)": "생선을 가끔 먹는 편이네요! 조금만 더 자주 먹으면 두뇌와 몸 건강에 더 도움이 돼요. 고등어나 연어처럼 오메가-3가 풍부한 생선을 일주일에 2~3번 먹는 것이 가장 좋아요. 생선구이, 생선조림, 연어샐러드 등 다양한 요리로 시도해보세요.",
            "3 (일주일에 2~3번 먹는다)": "아주 잘하고 있어요! 일주일에 2~3번 등푸른 생선을 먹으면 두뇌 발달, 눈 건강, 튼튼한 몸을 만드는 데 큰 도움이 돼요. 지금처럼 다양한 생선을 골고루 먹는 습관을 계속 이어가세요",
            "4 (일주일에 4번 이상 먹는다)": "생선을 자주 먹는 것도 좋지만, 너무 많이 먹으면 수은 같은 물질이 쌓일 수 있어요! 초등학생은 일주일에 2~3번, 한 번에 45~90g 정도가 적당해요. 생선을 너무 자주 먹는다면, 고기, 달걀, 콩 등 다른 단백질 식품과 번갈아 먹는 것도 중요해요. 건강을 위해 다양한 식품을 골고루 먹는 습관을 들이세요"
        }
    },
    
    {
        "text" : "하루 스마트폰 사용을 얼마나 하나요?",
        "options" : [
            "1 (1시간 이하)",
            "2 (2시간 초과 3시간 이하)",
            "3 (3시간 초과 5시간 이하)",
            "4 (5시간 초과)",
        ],
        "deduct" : {
            "1 (1시간 이하)": 0,
            "2 (2시간 초과 3시간 이하)" : 4,
            "3 (3시간 초과 5시간 이하)" : 8,
            "4 (5시간 초과)" : 10,
        },
        "advice" : {
            "1(1시간 이하)" : "아주 잘하고 있어요! 스마트폰을 하루 1시간 이내로 사용하면 눈 건강, 집중력, 자세 모두 지킬 수 있어요. 남는 시간에는 친구와 뛰어놀거나 책을 읽고, 취미 활동을 해보세요. 건강한 습관을 계속 유지하면 똑똑한 두뇌와 튼튼한 몸을 가질 수 있답니다!",
            "2 (2시간 초과 3시간 이하)" : "조금 길게 사용하는 편이에요! 스마트폰을 오래 보면 눈이 피로해지고, 집중력도 떨어질 수 있어요. 하루 1시간 이내로 줄이고, 사용할 때는 20분마다 눈을 쉬게 해주세요. 스마트폰 대신 운동이나 독서, 가족과 대화하는 시간을 늘려보면 더 건강해질 수 있어요!",
            "3(3시간 초과 5시간 이하)" : "스마트폰 사용 시간이 많아요! 하루 3~4시간 이상 사용하면 스트레스, 우울감, 수면장애, 눈 건강 문제, 자세 이상 등 여러 건강 문제가 생길 수 있어요. 하루 1시간 이내로 줄이고, 틈틈이 스트레칭을 하며, 스마트폰 대신 야외 활동이나 취미 생활을 해보세요. 부모님과 함께 사용 시간을 정해보는 것도 좋아요!",
            "4(5시간 초과)" : "스마트폰을 너무 오래 사용하면 건강에 큰 해가 될 수 있어요! 스트레스, 우울감, 집중력 저하, 수면장애, 눈 건강 악화, 목·어깨 통증, 심지어 중독 위험까지 높아져요. 하루 1시간 이내로 줄이고, 사용 시간에는 20분마다 눈을 쉬게 하세요. 스마트폰 대신 친구와 놀기, 운동, 독서 등 다양한 활동을 해보면 몸과 마음이 훨씬 건강해질 거예요! 부모님과 함께 사용 계획을 세워보는 것도 꼭 실천해보세요.",
        }
        
    },
    {
        "text" : "하루 중 스트레스를 얼마나 받나요?",
        "options" : [
            "1 (거의 없음)",
            "2 (조금 받는다)",
            "3 (많이 받는다)",
            "4 (매우 많이 받는다)",
        ],
        "deduct" : {
            "1 (거의 없음)": 0,
            "2 (조금 받는다)": 4,
            "3 (많이 받는다)": 8,
            "4 (매우 많이 받는다)": 10,
        },
        "advice" : {
            "1 (거의 없음)" : "정말 멋져요! 스트레스를 거의 받지 않는다면 마음이 건강하다는 뜻이에요. 지금처럼 긍정적으로 생각하고, 친구들과 잘 어울리며, 취미 활동도 즐기면 스트레스를 잘 이겨낼 수 있어요. 앞으로도 밝고 건강하게 지내길 응원할게요!",
            "2 (조금 받는다)" : "가끔 스트레스를 받는 건 누구나 그래요! 스트레스를 받을 때는 깊게 숨을 쉬거나, 좋아하는 음악을 듣고, 운동이나 산책을 해보세요. 가족이나 친구에게 마음을 털어놓는 것도 큰 도움이 돼요. 스트레스를 잘 풀면 건강한 하루를 보낼 수 있답니다!",
            "3(많이 받는다)" : "스트레스를 많이 받는다면, 혼자 고민하지 말고 꼭 주변에 이야기해보세요! 운동, 그림 그리기, 일기 쓰기, 명상 등 자신만의 스트레스 해소법을 찾아보는 것도 좋아요. 스트레스를 오래 쌓아두면 건강에 안 좋으니, 마음이 힘들 때는 부모님이나 선생님께 꼭 도움을 청하세요.",
            "4(매우 많이 받는다)" : "스트레스를 너무 많이 받으면 몸과 마음이 모두 힘들어질 수 있어요! 혼자 참지 말고 꼭 가족, 친구, 선생님께 이야기해보세요. 규칙적인 운동, 충분한 수면, 건강한 식사도 스트레스 해소에 도움이 돼요. 필요하다면 전문가의 도움을 받아보는 것도 괜찮아요. 무엇보다 내 마음이 가장 소중하다는 걸 꼭 기억하세요!",
        },
        
    },
    {
        "text" : "하루에 평균적으로 운동을 얼마나 많이 하나요?",
        "options" : [
            "1 (주 0회)",
            "2 (주 1~2회)",
            "3 (주 3~4회)",
            "4 (주 5~7회)",
        ],
        "deduct" : {
            "1 (주 0회)": 0,
            "2 (주 1~2회)": 4,
            "3 (주 3~4회)": 8,
            "4 (주 5~7회)": 10,
        },
        "advice" : {
            "1 (주 0회)" : "운동을 거의 하지 않으면 건강에 여러 가지 문제가 생길 수 있어요! 세계보건기구(WHO)에서는 5~17세 어린이와 청소년이 하루에 1시간(60분) 이상 운동할 것을 권장하고 있어요. 운동을 하면 키 성장, 뼈와 근육 강화, 집중력 향상, 스트레스 해소에 모두 도움이 돼요. 오늘부터라도 걷기, 줄넘기, 축구, 자전거 타기 등 재미있는 운동을 시작해보세요! 친구들과 함께하면 훨씬 즐겁게 할 수 있답니다.",
            "2 (주 1~2회)" : "조금 더 자주 운동하면 더 건강해질 수 있어요! 일주일에 3번 이상 운동하는 것이 가장 좋아요. 운동을 하면 몸도 튼튼해지고, 공부할 때 집중력도 좋아져요. 운동하는 날을 정해두고, 가족이나 친구와 함께 꾸준히 해보세요!",
            "3(주 3~4회)" : "좋은 습관이에요! 일주일에 3~4번 운동하면 건강을 잘 지킬 수 있어요. 다양한 운동(달리기, 줄넘기, 농구, 수영 등)을 번갈아 해보면 더 재미있고 효과도 좋아요. 앞으로도 꾸준히 운동하는 멋진 습관을 계속 이어가길 응원할게요!",
            "4(주 5~7회)" : "정말 멋져요! 운동을 자주 하면 키도 쑥쑥, 몸도 튼튼해져요! 지금처럼 일주일에 5번 이상 운동하면 체력, 집중력, 면역력 모두 좋아져요. 너무 무리하지 않도록 쉬는 날도 꼭 챙기고, 운동 전후로 스트레칭과 물 마시기를 잊지 마세요. 앞으로도 건강한 운동 습관을 계속 유지해보세요!",
        },
    }
]

class HealthSurveyApp(QtWidgets.QMainWindow):
    def __init__(self):
        super().__init__()
        uic.loadUi("sample.ui", self)

        self.current_index = 0
        self.score = 100
        self.selected_answers = [None] * len(questions)

        # 위젯 연결
        self.btnNext.clicked.connect(self.next_question)
        self.btnPrev.clicked.connect(self.prev_question)

        # 라디오버튼들 리스트로
        self.option_buttons = [
            self.radioOption1,
            self.radioOption2,
            self.radioOption3,
            self.radioOption4,
        ]

        self.load_question()

    def load_question(self):
        q = questions[self.current_index]

        # 문제 번호 표시
        self.lblQuestionNumber.setText(f"문제 {self.current_index + 1} / {len(questions)}")

        # 문제 텍스트
        self.lblQuestionText.setText(q["text"])

        # 옵션 라디오 버튼 세팅
        for i, btn in enumerate(self.option_buttons):
            if i < len(q["options"]):
                btn.setText(q["options"][i])
                btn.show()
                btn.setChecked(False)
                btn.setEnabled(True)
            else:
                btn.hide()

        # 이전에 선택한 답 복원
        prev_answer = self.selected_answers[self.current_index]
        if prev_answer:
            for btn in self.option_buttons:
                if btn.text() == prev_answer:
                    btn.setChecked(True)
                    break

        # 버튼 활성화 여부
        self.btnPrev.setEnabled(self.current_index > 0)
        if self.current_index == len(questions) - 1:
            self.btnNext.setText("결과 보기")
        else:
            self.btnNext.setText("다음 문제")

    def next_question(self):
        # 선택한 답 체크
        selected = None
        for btn in self.option_buttons:
            if btn.isChecked():
                selected = btn.text()
                break

        if not selected:
            QtWidgets.QMessageBox.warning(self, "알림", "답변을 선택해 주세요!")
            return

        # 선택 답 저장
        self.selected_answers[self.current_index] = selected

        # 마지막 문제면 결과 표시
        if self.current_index == len(questions) - 1:
            self.calculate_and_show_result()
            return

        # 다음 문제로 이동
        self.current_index += 1
        self.load_question()

    def prev_question(self):
        if self.current_index > 0:
            self.current_index -= 1
            self.load_question()

    def calculate_and_show_result(self):
        total_deduct = 0
        advice_list = []

        for idx, answer in enumerate(self.selected_answers):
            q = questions[idx]
            deduct_score = q["deduct"].get(answer, 0)
            total_deduct += deduct_score
            if deduct_score > 0:
                advice_list.append(f"문제 {idx + 1}: {q['advice'][answer]}")

        final_score = max(0, 100 - total_deduct)

        advice_text = "\n\n".join(advice_list) if advice_list else "모든 문항에서 좋은 상태입니다!"

        msg = f"최종 점수: {final_score}점\n\n감점 사유 및 조언:\n{advice_text}"

        QtWidgets.QMessageBox.information(self, "건강 설문 결과", msg)

app = QtWidgets.QApplication(sys.argv)
window = HealthSurveyApp()
window.show()
sys.exit(app.exec_())
